plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/cache/CachingHiveMetastore.java

```diff
@@ -271,19 +272,19 @@ protected CachingHiveMetastore(HiveMetastore delegate, OptionalLong expiresAfter
         grantedPrincipalsCache = cacheFactory.buildCache(this::loadPrincipals);
         configValuesCache = cacheFactory.buildCache(this::loadConfigValue);
 
-        partitionStatisticsCache = partitionCacheFactory.buildCache(this::loadPartitionColumnStatistics, this::loadPartitionsColumnStatistics);
+        partitionStatisticsCache = partitionCacheFactory.buildBulkCache(this::loadPartitionsColumnStatistics);
         partitionFilterCache = partitionCacheFactory.buildCache(this::loadPartitionNamesByFilter);
-        partitionCache = partitionCacheFactory.buildCache(this::loadPartitionByName, this::loadPartitionsByNames);
+        partitionCache = partitionCacheFactory.buildBulkCache(this::loadPartitionsByNames);
     }
 
     private static <K, V> LoadingCache<K, V> neverCache(com.google.common.base.Function<K, V> loader)
     {
         return buildCache(OptionalLong.of(0), OptionalLong.empty(), Optional.empty(), 0, StatsRecording.DISABLED, loader);
     }
 
-    private static <K, V> LoadingCache<K, V> neverCache(Function<K, V> loader, Function<Iterable<K>, Map<K, V>> bulkLoader)
+    private static <K, V> LoadingCache<K, V> neverCache(Function<Iterable<K>, Map<K, V>> bulkLoader)
```

Comment:
There are now two `neverCache(Function)` methods. It's not allowed in Java due to type erasure. It works here only because one is `com.google.common.base.Function` and the other is `java.util.function.Function`, which is a very clever hack around the Java limitation.

Please rename this one to eg `neverCacheBulk`

Reply:
> Please rename this one to eg `neverCacheBulk`

Actually, it would be more readable to remove these methods, see https://github.com/trinodb/trino/pull/15846

Reply:
I see this resolved, but I don't see what is the resolution. What is it?

Reply:
BTW https://github.com/trinodb/trino/pull/15846 was meant to help resolve this discussion, i.e. avoid having to make a funny name for a method that can simply be deleted. Up to you though

Reply:
I thought you wanted to merge https://github.com/trinodb/trino/pull/15846 after this PR